### PR TITLE
[Migration][DO NOT MERGE] Separate old ir into `_legacy_ir` folder

### PR DESCRIPTION
**Pull request by BowenBao**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #1334
* #1340
* #1345
* #1343
* #1333
* __->__ #1332
* #1331
* #1330
* #1329
* #1328

All tests except for linter are expected to pass.

### CODE REVIEW

**Comment by justinchuby** [comment]
Be sure to remove reference from ir/__init__.py as well?

File: onnxscript/ir/__init__.py [hunk 1]
```diff
@@ -1,6 +1,4 @@
 """In-memory intermediate representation for ONNX graphs."""
-from __future__ import annotations
```
Comment (justinchuby) [inline onnxscript/ir/__init__.py:2]: Keep?

**Review by justinchuby — APPROVED** [review]

**Review by justinchuby — APPROVED** [review]

File: onnxscript/ir/__init__.py (hunk 1 above)
Comment (BowenBao) [reply onnxscript/ir/__init__.py:2]: was introduced by merging w/ `__init__.py` from legacy ir. If needed we can add in follow ups.

**Comment by BowenBao** [comment]
Close as squashed and merged into main.

**Pull request closed without merging**